src/main/java/com/orm/session/Session.java

```diff
@@ -37,10 +45,39 @@ public <T> T find(Class<T> entityType, Object id) {
         return entityType.cast(result);
     }
 
-    public void close() {
-        updateIfChanged();
-        cacheMap.clear();
+    public void persist(Object entity) {
+        actionQueue.add(new InsertAction(entity));
+    }
+
+    public void remove(Object entity) {
+        actionQueue.add(new DeleteAction(entity));
+    }
+
+    public void flush() {
+//        updateIfChanged();
         snapshotMap.clear();
+        executeActions();
+        cacheMap.clear();
+    }
```

Comment:
so far as I remember, `flush` only synchronizes the state between Session and DB and doesn't clear the 1st level cache